### DISCUSSION TITLE
Revapi not failing after removing ignore extension

### DISCUSSION THREAD

**Discussion by triceo** [revapi/revapi#290] [Q&A]
Hey there.

I have a Revapi config like so:

        <plugin>
          <groupId>org.revapi</groupId>
          <artifactId>revapi-maven-plugin</artifactId>
          <dependencies>
            <dependency>
              <groupId>org.revapi</groupId>
              <artifactId>revapi-java</artifactId>
            </dependency>
          </dependencies>
          <configuration>
            <checkDependencies>false</checkDependencies>
            <oldArtifacts>
              <artifact>${project.groupId}:${project.artifactId}:${revapi.oldTimefoldVersion}</artifact>
            </oldArtifacts>
            <newArtifacts>
              <artifact>${project.groupId}:${project.artifactId}:${project.version}</artifact>
            </newArtifacts>
            <analysisConfigurationFiles>
              <file>${project.basedir}/src/build/revapi-filter.json</file>
            </analysisConfigurationFiles>
            <reportCriticality>documented</reportCriticality>
            <failCriticality>highlight</failCriticality>
          </configuration>
        </plugin>

I used to use the `ignore` extension, but now I removed it, keeping only the `filter`. To my surprise, Revapi no longer fails on any differences found.
Yet, there clearly seem to be some differences to fail on, as evidenced by Maven's debug logging:

    transform-block [org.revapi.java.transforms.annotations.NowDeprecated@75dff25d]: {occurrences = 11, total = 1ms, average = 0,09ms, worstTime = 1ms caused by Difference[code='java.annotation.added', name='new annotation', classification={SEMANTIC=POTENTIALLY_BREAKING, SOURCE=EQUIVALENT, BINARY=EQUIVALENT}, description='Element newly annotated with 'java.lang.Deprecated'.', justification='null', criticality=null, attachments={annotationType=java.lang.Deprecated, annotation=@java.lang.Deprecated(forRemoval = true, since = "1.1.0"), package=ai.timefold.solver.core.api.domain.valuerange, classQualifiedName=ai.timefold.solver.core.api.domain.valuerange.ValueRangeFactory, classSimpleName=ValueRangeFactory, methodName=createDoubleValueRange, elementKind=method, oldArchive=ai.timefold.solver:timefold-solver-core-impl:jar:0.9.38, oldArchiveRole=primary, newArchive=ai.timefold.solver:timefold-solver-core-impl:jar:1.2.0-SNAPSHOT, newArchiveRole=primary}]}

In fact, no differences are even reported, even though `reportCriticality` is set fairly liberally.

What am I doing wrong?

**Comment by metlos** [comment]
You got me worried here, but I hope things are more or less working properly.

You have configured the criticality to fail the build to `highlight` which means all potentially breaking builds should break the build. But as far as I checked you don't have any - by a quick look, you only seem to have equivalent changes (like deprecations) and non-breaking changes.

If you change the `failCriticality` to `allowed`, you will fail the build and it will list all the found API changes.

When I tried to run `mvn revapi:report`, you are right that the `target/site/revapi-report.html` doesn't contain any changes, even though I would also have guessed it should contain the non-breaking changes. Not sure why that is, but I haven't had time to investigate that further.

For reporting, if you're not using `mvn site`, I would suggest using some revapi-specific reporter like https://revapi.org/revapi-reporter-text/0.15.0/index.html or https://revapi.org/revapi-reporter-json/0.5.0/index.html that produce more machine friendly output...

Just wanted to let you know not all is broken ;)
